### DISCUSSION TITLE
Using Bonsai to record a video of certain time long

### DISCUSSION THREAD

**Discussion by Sigmalalala** [bonsai-rx/bonsai#1140] [Q&A]
Hello，I'm a new Bonsai user and i want to record a video for a certain length such as 1 or 5 minutes, but I have to set the framerate in the "VideoWriter". In this case, the video I get is not exactly  1 or 5 minutes, because the frame rate of my camera is changing at all times. So, I wonder if there is a better work flow or any suggestions. Thank you so much!
![1668323583505](https://user-images.githubusercontent.com/118099060/201511170-294ba8a9-4881-43c7-b733-adea6c3232d0.jpg)
![1668323963435](https://user-images.githubusercontent.com/118099060/201511178-b6e955b1-9327-45e0-a51c-8883edc52d5c.jpg)

**Comment by bruno-f-cruz** [comment]
You can try to stabilize the fps of your camera by turning off the auto-exposure setting. Afterward, you should be able to get something close to the predicted duration from your camera fps and videowriter settings.

**Reply by Sigmalalala** [reply]
Thank you very much for your apply! I have tried that before, but the fps still changed between 33 and 34. So every time i want to get a video of 5 minutes, but the result is that i get a vedio of 5minutes and 1 or 2 seconds. Any other suggestions? Thank you so much again!

**Reply by bruno-f-cruz** [reply]
IF you REALLY want to force the duration of the video simply use the "Take(N)" operator to get N number of frames.
e.g. if your videowriter is set at 30fps, 30fps*10seconds = 300 frames.

**Reply by Sigmalalala** [reply]
Sorry, but I don't know how to use the "take" operator. Could you please show the workflow to me? Thank you!
![1668337381493](https://user-images.githubusercontent.com/118099060/201518438-8e20fc77-67e7-4c75-b728-2a636be82ec7.jpg)
My workflow is absolutuely wrong.

**Comment by bruno-f-cruz** [comment]
CameraCapture - Take(N frames) - videowriter
That’s it. You can remove the other two nodes.

**Reply by Sigmalalala** [reply]
I use this "Take" operator. Am I right? But I don't know how to take the frames. Could you please tell me how to take frames? Thank you!
![1668337977908](https://user-images.githubusercontent.com/118099060/201518768-0cdff14d-2c2f-4452-8bc8-333b033b13b6.jpg)

![1668337911165](https://user-images.githubusercontent.com/118099060/201518725-1d33479f-8a17-4bf1-8b41-85bddfe143ac.jpg)

**Comment by bruno-f-cruz** [comment]
Change the count property to the number of frames you want to save

**Reply by Sigmalalala** [reply]
I tried it. But how can i add "timer" operator? I want to record a video of certain time long such as 1 minutes. This workflow doesn't work. Can i make a friend with you on wechat? My wechat number:Sigma0089. Thank you so much.
![1668337381493](https://user-images.githubusercontent.com/118099060/201519768-6ba1246e-322f-4c34-a040-133fd5e14284.jpg)

**Reply by bruno-f-cruz** [reply]
I thought you wanted to have a video with a fixed duration and fixed fps? If that is the case, the way to achieve this is to record N frames where N = fps * duration(seconds). If you want to define a timer you will always go back to the problem you were having before.

**Reply by Sigmalalala** [reply]
I make it. Thank you very much! So nice of you.
